src/device/wg_controller.rs

```diff
@@ -219,15 +220,81 @@ async fn consolidate_wg_peers<
         aggregator.report_peer_state_relayed(&event).await;
 
         wireguard_interface.del_peer(*key).await?;
+
+        // Remove from session_keeper
+        if let Some(sk) = session_keeper {
+            sk.remove_node(key).await?;
+        }
     }
 
-    for key in insert_keys {
-        let requested_peer = requested_peers.get(key).ok_or(Error::PeerNotFound)?;
-        telio_log_info!("Inserting peer: {:?}", requested_peer);
+    let batcher_threshold = features
+        .batching
+        .as_ref()
+        .map(|b| Duration::from_secs(b.threshold as u64));
+
+    let get_peer = |key| -> Result<(Peer, Option<Duration>)> {
+        if features.batching.is_some() {
+            // If batching is enabled, we use the session keeper for the persistent keepalives
+            let mut peer = requested_peers
+                .get(key)
+                .ok_or(Error::PeerNotFound)?
+                .peer
+                .clone();
+
+            // Persistent keepalive interval has been set by the build_requested_peers_list for
+            // every node. It considers type of connection proxy/direct and special nodes as VPN or STUN.
+            // We want to disable WG keepalives and use our session keeper instead
+            telio_log_debug!(
+                "Keepalive interval for {:?} is {:?}",
+                peer,
+                peer.persistent_keepalive_interval
+            );
+            let keepalive_interval = Duration::from_secs(
+                peer.persistent_keepalive_interval
+                    .map(|k| {
+                        if k == 0 {
+                            requested_state.keepalive_periods.direct
+                        } else {
+                            k
+                        }
+                    })
+                    .unwrap_or(requested_state.keepalive_periods.direct)
+                    .into(),
+            );
+            peer.persistent_keepalive_interval = Some(0); // Disable WG persistent keepalive
```

Comment:
I am afraid that reusing persistent_keepalive_interval in `RequestedPeer` structure may be a bit confusing, if we are just simply going to override it in all cases to `0` later. Maybe lets add a separate field dedicated for holding just the session_keeper period?